opencti-platform/opencti-graphql/src/migrations/migration-threat-actors-relationships.js

```diff
@@ -0,0 +1,246 @@
+    filters: [
+      {
+        key: 'fromTypes',
+        mode: 'or',
+        operator: 'eq',
+        values: [
+          'Threat-Actor-Group'
+        ]
+      },
+      {
+        key: 'toTypes',
+        mode: 'or',
+        operator: 'eq',
+        values: [
+          'Threat-Actor-Individual'
+        ]
+      }
+    ],
+    mode: 'and'
+  };
+  // We get all current part-of & derived-from relations: will be used to check for duplicates in part-of/derived-from to related-to migration
+  const getTAIPartOfRelation = await elList(context, SYSTEM_USER, READ_INDEX_STIX_CORE_RELATIONSHIPS, {
+    types: [RELATION_PART_OF],
+    filters: taiTargetTaiFilter
+  });
+  const getTAGPartOfRelation = await elList(context, SYSTEM_USER, READ_INDEX_STIX_CORE_RELATIONSHIPS, {
+    types: [RELATION_PART_OF],
+    filters: tagTargetTaiFilter
+  });
+  const getTAIDerivedFromRelation = await elList(context, SYSTEM_USER, READ_INDEX_STIX_CORE_RELATIONSHIPS, {
+    types: [RELATION_DERIVED_FROM],
+    filters: taiTargetTaiFilter
+  });
+
+  const invalidRelationships = [...getTAIDerivedFromRelation, ...getTAIPartOfRelation, ...getTAGPartOfRelation];
+  // If no invalidRelationships exist, we can skip the migration process
+  if (invalidRelationships.length === 0) {
+    logApp.info('[MIGRATION] Update all invalid relationships between threat-actors: no relationships to migrate');
+    next();
+    return;
+  }
+  logApp.info(`[MIGRATION] Update all invalid threat-actors relationships to related-to: ${invalidRelationships.length} relationships to migrate`);
+
+  const searchRelatedToDuplicateFn = async (from, to) => {
+    const countDuplicateQueryBody = {
+      bool: {
+        must: [
+          { term: { 'entity_type.keyword': { value: RELATION_RELATED_TO } } },
+          {
+            nested: {
+              path: 'connections',
+              query: {
+                bool: {
+                  must: [
+                    { term: { 'connections.role.keyword': { value: 'related-to_from' } } },
+                    { term: { 'connections.internal_id.keyword': { value: from } } }
+                  ]
+                }
+              }
+            }
+          },
+          {
+            nested: {
+              path: 'connections',
+              query: {
+                bool: {
+                  must: [
+                    { term: { 'connections.role.keyword': { value: 'related-to_to' } } },
+                    { term: { 'connections.internal_id.keyword': { value: to } } }
+                  ]
+                }
+              }
+            }
+          }
+        ]
+      }
+    };
+    const findQuery = {
+      index: READ_INDEX_STIX_CORE_RELATIONSHIPS,
+      body: { query: countDuplicateQueryBody },
+    };
+
+    const rawCount = await elRawCount(findQuery);
+    return rawCount > 0;
+  };
+
+  const invalidRelationsDuplicatesIds = [];
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < invalidRelationships.length; i++) {
```

Comment:
```suggestion
  for (let i = 0; i < invalidRelationships.length; i+=1) {
```
the fix is to use i+=1 instead of i++